tests/Microsoft.Bot.Builder.Tests/RootFrameTests.cs

```diff
@@ -0,0 +1,180 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Moq;
+
+namespace Microsoft.Bot.Builder.Tests
+{
+    [TestClass]
+    [TestCategory("State")]
+    public class RootFrameTests
+    {
```

Comment:
See other comment on hierarchical test pattern. Every set of tests for a particular method here should get a nested class to avoid the `MethodName_` prefixing... plays better with modern test runners.